appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockBarChartDataModel.java

```diff
@@ -0,0 +1,200 @@
+
+    // Adds the Data Series to the Chart.
+    addDataSeriesToChart();
+  }
+
+  @Override
+  public void changeColor(String color) {
+    color = getHexColor(color);
+    dataSeries.setBackgroundColor(color);
+    dataSeries.setBorderColor(color);
+  }
+
+  @Override
+  protected void setDefaultElements() {
+    final int points = 4; // Number of points to add
+
+    // Set the starting y value for the current model
+    // as the total size of the Datasets present in the Chart.
+    // This is used to differentiate values between added Data
+    // components.
+    // TODO: In the future, this could take into account the global
+    // TODO: maximum (much like the MockLineChartDataModel does now)
+    // TODO: or have recalculations done, since if Data Series are
+    // TODO: deleted, the y values can overlap between Data Series.
+    double yVal = chartData.getDatasets().size();
+
+    for (int i = 0; i < points; ++i) {
+      // Construct the x and y values based on the index
+      double yValue = yVal + i;
+
+      // Add an entry based on the constructed values
+      addEntryFromTuple((double) i, yValue);
+    }
+  }
+
+  @Override
+  protected void setDefaultStylingProperties() {
+
+  }
+
+  @Override
+  public void addEntryFromTuple(String... tuple) {
+    try {
+      // Parse x and y values
+      double xValue = Double.parseDouble(tuple[0]);
+      double yValue = Double.parseDouble(tuple[1]);
+
+      // Add an entry from the parsed values
+      addEntryFromTuple(xValue, yValue);
+    } catch (NumberFormatException e) {
+      // Wrong input. Do nothing.
+    }
+  }
+
+  /**
+   * Adds an entry to the Data Series from the specified tuple.
+   *
+   * The tuple is expected to have at least 2 entries. All subsequent
+   * values are ignored.
+   *
+   * The addEntryFromTuple method relies on the property that each
+   * entry in the Bar Chart corresponds to an index, meaning that
+   * an x value directly corresponds to an index in the Data Series.
+   * This also implies that the entries in the Data Series are
+   * sorted by x value. Moreover, if entries with an x value
+   * which is at least 1 bigger than the current size is added,
+   * blank entries are filled in between.
+   *
+   * @param tuple  tuple (array of doubles)
+   */
+  public void addEntryFromTuple(Double... tuple) {
+    // The first entry of the tuple is expected to represent
+    // the x value. Since the Bar Chart's x values are whole
+    // numbers, the value has to be rounded.
+    int xValue = (int) Math.round(tuple[0]);
+
+    // The second entry of the tuple is expected to be the y value.
+    double yValue = tuple[1];
+
+    // If x is less than 0, then skip the insertion, since
+    // Bar Chart x values start from 0.
+    if (xValue < 0) {
+      return;
+    }
+
+    // Same data adding logic is used as in BarChartDataModel
+    // (Android implementation)
+    // TODO: This could probably be factored somehow to reduce redundancy
```

Comment:
@ewpatton Fair enough; I guess we can leave the TODO here whatsoever though in case it is doable?
Also, will the iOS version require to change the mock components? If not, then would this have to be rewritten?